e2e.groovy

```diff
@@ -0,0 +1,62 @@
+def cico_retries = 16
+def cico_retry_interval = 60
+def ci_git_repo = 'https://github.com/Yuggupta27/ceph-csi'
+def ci_git_branch = 'e2e-centosci'
+def ref = "master"
+
+node('cico-workspace') {
+
+	stage('checkout ci repository') {
+		git url: "${ci_git_repo}",
+			branch: "${ci_git_branch}",
+			changelog: false
+	}
+
+	stage('reserve bare-metal machine') {
+		def firstAttempt = true
+		retry(30) {
+			if (!firstAttempt) {
+				sleep(time: 5, unit: "MINUTES")
+			}
+			firstAttempt = false
+			cico = sh(
+				script: "cico node get -f value -c hostname -c comment --retry-count ${cico_retries} --retry-interval ${cico_retry_interval}",
+				returnStdout: true
+			).trim().tokenize(' ')
+			env.CICO_NODE = "${cico[0]}.ci.centos.org"
+			env.CICO_SSID = "${cico[1]}"
+		}
+	}
+	try {
+		stage('prepare bare-metal machine') {
+			if ("${ghprbPullId}".length() != 0) {
+				ref = "pull/${ghprbPullId}/head"
+			}
+			sh 'scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ./prepare.sh ./multi-node-k8s.sh root@${CICO_NODE}:'
+			sh "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./prepare.sh --workdir=/opt/build/go/src/github.com/ceph/ceph-csi --gitrepo=${ci_git_repo} --ref=master"
+		}
+
+		stage('deploy kubernetes') {
+			sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./multi-node-k8s.sh'
```

Comment:
After deploying Kubernetes in the VMs, it is needed to build the ceph-csi container image and push it to the VMs. Please mark this as a TODO. Without locally building the ceph-csi image, it will get pulled from Quay.io (latest master build) and the changes of the PR will not get tested.